external_modules/src/peanut_moveit_interface/src/moveit_interface.py

```diff
@@ -0,0 +1,86 @@
+#! /usr/bin/env python
+import rospy
+from util import *
+import numpy as np
+
+import moveit_commander
+import moveit_msgs.msg
+from geometry_msgs.msg import Pose
+
+from peanut_moveit_interface.srv import *
+
+
+class MoveitInterface(object):
+    """Interface between higher lvl code operating through botx and moveit
+    requires moveit to be launched and controllers to be up
+    """
+
+    def __init__(self):
+
+        # this variable is set when a service is being handled
+        self.processing_srv = False
+
+        # start the moveit commander
+        moveit_commander.roscpp_initialize(sys.argv)
+
+        # create robot commander instance
+        self.robot = moveit_commander.RobotCommander()
+
+        # initialize the planning scene
+        self.scene = moveit_commander.PlanningSceneInterface()
+
+        # initialize the arm move group
+        self.group = moveit_commander.MoveGroupCommander("arm")
+
+        # for publishing trajectories to Rviz for visualization
+        display_trajectory_publisher = rospy.Publisher('/move_group/display_planned_path',
+                                                       moveit_msgs.msg.DisplayTrajectory,
+                                                       queue_size=20)
+
+        # TODO replace with an action service once botx can handle that
+        rospy.Service('peanut_moveit_interface', MoveitInterfaceGoal, self.higherup_handler)
+
+    def execute_pose(self, pose_msg):
+        """
+        send the pose to moveit
+        :param pose_msg: geometry_msgs/Pose
+        :return string success or failed (for now):
```

Comment:
is this comment compliant with an automatic documentation generation system?

Reply:
TB discussed